pkg/leap/leap-file.go

```diff
@@ -58,19 +62,42 @@ type LeapFile struct {
 	Hash           string      `json:"hash"`
 }
 
-func New(kubeclient *kubernetes.Clientset, namespace string) (*LeapManager, error) {
-	lm := &LeapManager{
-		UbloxLsInd: make(chan ublox.TimeLs, 2),
-		Close:      make(chan bool),
-		client:     kubeclient,
-		namespace:  namespace,
-		leapFile:   LeapFile{},
+var lock = &sync.Mutex{}
+
+var leapManager *LeapManager
+
+func New(kubeclient kubernetes.Interface, namespace string) (*LeapManager, error) {
```

Comment:
Make leap manager a singletone that can provide service for anyone, without a need to pass a pointer to the instance